src/connection.ts

```diff
@@ -16,12 +25,70 @@ import {
   RequestInfo,
   Tooling as JSForceTooling,
 } from 'jsforce';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import * as HttpApi from 'jsforce/lib/http-api';
 import { AuthFields, AuthInfo } from './authInfo';
 import { ConfigAggregator } from './config/configAggregator';
 import { Logger } from './logger';
 import { SfdxError } from './sfdxError';
 import { sfdc } from './util/sfdc';
 
+// export class PerformanceMetrics {
+//   private metrics: PerformanceMetrics.Metric[] = [];
+
+//   public add(metric: PerformanceMetrics.Metric): void {
+//     this.metrics.push(metric);
+//   }
+
+//   public collect(): PerformanceMetrics.Metric[] {
+//     const copy = this.copy();
+//     this.clear();
+//     return copy;
+//   }
+
+//   private copy(): PerformanceMetrics.Metric[] {
+//     return this.metrics.map((m) => m);
+//   }
+
+//   private clear(): void {
+//     this.metrics = [];
+//   }
+// }
+
+// namespace PerformanceMetrics {
+//   export interface Metric {
+//     requestPath: string;
+//     perfMetrics: AnyJson;
+//   }
+
+//   export interface Response {
+//     headers: AnyJson & { perfmetrics?: string };
+//     req: { path: string };
+//   }
+// }
+
+interface Response {
+  headers: AnyJson & { perfmetrics?: string };
+  req: { path: string };
+}
+
+// Unfortunately, we have to monkey patch HttpApi.prototype.request in order
```

Comment:
I thought we talked about having this contained in the data plugin for now? Just have a `DataConnection extends Connection` class in the data plugin to do all this. For the auth info, you can always just create a new one if you need to.

Reply:
@amphro overriding the jsforce function doesn't work if I do it from plugin-data